event-based-simulators/extras/ExportProfileData.py

```diff
@@ -0,0 +1,233 @@
+    if not externalIdResponse.ok:
+        logger.error(
+            f"Connection to url '{C8Y_BASE}/identity/globalIds/{deviceId}/externalIds' failed. Check your parameters in environment file again")
+        sys.exit()
+    else:
+        return externalIdResponse
+
+
+def checkDeviceExternalIdById(deviceId):
+    externalIdResponse = getExternalIdReponse(deviceId)
+
+    try:
+        deviceExternalId = externalIdResponse.json()['externalIds'][0]['externalId']
+        deviceExternalIdType = externalIdResponse.json()['externalIds'][0]['type']
+    except:
+        logger.error(f"Could not find external id for the device with id {deviceId}")
+        print(f"Could not find external id for the device with id {deviceId}")
+        return None, None
+
+    return deviceExternalId, deviceExternalIdType
+
+
+def isExternalIdHasEventBasedSimulatorProfileType(deviceExternalIdType):
+    if deviceExternalIdType == "c8y_EventBasedSimulatorProfile":
+        return True
+    else:
+        logger.debug(f"The type {deviceExternalIdType} of external ID must be c8y_EventBasedSimulatorProfile")
+        print(f"The type {deviceExternalIdType} of external ID must be c8y_EventBasedSimulatorProfile")
+        return False
+
+
+def createFilePath(fileName):
+    # Check if folder containing data files exists and make one if not
+    if not os.path.exists('export_data'):
+        os.makedirs('export_data')
+    relativeFilePath = f'export_data\{fileName}.json'
+    filePath = os.path.join(os.path.dirname(__file__), relativeFilePath)
+    logger.info(f"Created successfully file path: {filePath}")
+    return filePath
+
+
+def SetTimePeriodToExportData(CREATE_FROM, CREATE_TO):
+    if not CREATE_FROM or CREATE_TO:
+        createTo = datetime.now().replace(tzinfo=timezone.utc)
+        TimeUnit = Environment.TIME_UNIT
+
+        if TimeUnit == 'seconds' or not TimeUnit:
+            createFrom = createTo - timedelta(seconds=Environment.PERIOD_TO_EXPORT)
+            return createFrom, createTo
+
+        if TimeUnit == 'days':
+            createFrom = createTo - timedelta(days=Environment.PERIOD_TO_EXPORT)
+        elif TimeUnit == 'weeks':
+            createFrom = createTo - timedelta(weeks=Environment.PERIOD_TO_EXPORT)
+        elif TimeUnit == 'hours':
+            createFrom = createTo - timedelta(hours=Environment.PERIOD_TO_EXPORT)
+        elif TimeUnit == 'minutes':
+            createFrom = createTo - timedelta(minutes=Environment.PERIOD_TO_EXPORT)
+        return createFrom, createTo
+
+    return CREATE_FROM, CREATE_TO
+
+
+def removeSlashFromBaseUrl():
+    if Environment.C8Y_BASE[-1] == '/':
+        C8Y_BASE = Environment.C8Y_BASE[:-1]
+    else:
+        C8Y_BASE = Environment.C8Y_BASE
+    return C8Y_BASE
+
+
+# Main function to run the script
+if __name__ == '__main__':
+    c8y = ArgumentsAndCredentialsHandler.c8yPlatformConnection()
+    # Check if connection to tenant can be created
+    C8Y_BASE = removeSlashFromBaseUrl()
```

Comment:
You even called it at the beginning. So i think you can remove it from every method